sno/ogr_import_source.py

```diff
@@ -624,16 +629,15 @@ def features(self):
         Overrides the super implementation for performance reasons
         (it turns out that OGR feature iterators for GPKG are quite slow!)
         """
-        db = gpkg.db(self.ogr_source)
-        dbcur = db.cursor()
-        dbcur.execute(f"SELECT * FROM {self.quote_ident(self.table)};")
-        for gpkg_feature in dbcur:
-            yield self._gpkg_feature_to_sno_feature(gpkg_feature)
+        with self.engine.connect() as db:
+            r = db.execute(f"SELECT * FROM {self.quote_ident(self.table)};")
```

Comment:
I won't switch all the statements to be the sqlalchemy-domain-specific-language yet. Except I did the ones for inserting to the user-defined-tables already, but that was actually because it didn't work otherwise due to how sqlalchemy placeholders work.
There'll be plenty of them to do.